text/0965-reactivity.md

```diff
@@ -0,0 +1,419 @@
+
+Additionally, this was taken as an opportunity to move from push-based reactivity to pull-based reactivity, and lead to a long exploration of patterns, trial-and-error, and developing new patterns to achieve easier testing, debugging and maintenance.  But most of this happened long after the official documentation for Octane had landed -- and even at the time of writing this RFC, the most important concepts about reactivity would fall well under the ["In-depth topics"](https://guides.emberjs.com/release/in-depth-topics/autotracking-in-depth/) section of the documentation. 
+
+This RFC aims for two things:
+- focus on teaching reactivity concepts, so that folks can be better equipped for what would be "advanced concepts" today.
+- defining and bringing cohesion to our reactive primitives so that the above is much easier to achieve. 
+  - This may necessitate additional sub-RFCs. 
+
+## Motivation
+
+The ember ecosystem has a number of "third-party" abstractions for some core reactive primitives, that could benefit (from teaching, performance, and cohesion perspectives) from being pulled in to a cohesive set of utilities from one source. This will help with longer-term goals of [simplifying imports](https://github.com/emberjs/rfcs/pull/946) -- today we have a _pit of exhaustion_ when it comes to how many imports are needed to make components.
+
+
+We have an opportunity to improve a lot of performance in how ember renders by collapsing abstractions. 
+
+The "third-party" libraries are:
+- `ember-modifier`, providing: `modifier`
+- `ember-resources`, providing: `resource`, `cell`
+- `tracked-built-ins`, providing: `TrackedArray`, `TrackedObject`, etc 
+
+There are others, but these are the most commonly used. `@glimmer/*` was meant to be the _foundation_ upon which ember was built, and it still provides that function. Since we've introduced modifier managers, and helper managers, we've enabled a healthy amount of experimentation, and it's time to reign it all in under a cohesive story.
+
+
+We want to distill concepts so that folks coming from other ecosystems, can more easily adopt, and learn one thing at a time as fast or as slow as they want.
+
+<detail><summary>two things not mentioned here</summary>
+
+1. helpers - since ember-source 4.5 (3.25 with the [polyfill](https://github.com/ember-polyfills/ember-functions-as-helper-polyfill)), helpers are a concept of the past. We can use regular functions -- 0 user-land abstraction.  However, changes to the default helper manager will be required to support `modifier` and `resource`, when used in a certain way (preview: this will be called "collapsing")
+
+2. components - components are not a primitive, and moreso a refactoring boundary -- they combine the above things into packaged micro-bundles.
+
+</details>
+
+## Detailed design
+
+tl;dr:
+- define primitives
+- define integration points
+- manager pattern is fallback, rather than default
+
+[starbeam]: https://www.starbeamjs.com/
```

Comment:
It's not a vital or an important change but, to keep things consistent, should this "link-reference"
1. be moved into the section where it is first introduced? (just below, `### Preface`)
2. at the bottom of the file?